tools/installpsh-amazonlinux.sh

```diff
@@ -0,0 +1,221 @@
+#bash <(curl -s https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/installpsh-amazonlinux.sh) <ARGUMENTS>
+
+#Usage - if you do not have the ability to run scripts directly from the web, 
+#        pull all files in this repo folder and execute, this script
+#        automatically prefers local copies of sub-scripts
+
+#Completely automated install requires a root account or sudo with a password requirement
+
+#Switches
+#  -includeide - the script is being run headless, do not perform actions that require response from the console
+#  -interactivetests - requires a human user in front of the machine - loads a script into the ide to test with F5 to ensure the IDE can run scripts
+
+#gitrepo paths are overrideable to run from your own fork or branch for testing or private distribution
+
+
+VERSION="1.1.2"
+gitreposubpath="PowerShell/PowerShell/master"
+gitreposcriptroot="https://raw.githubusercontent.com/$gitreposubpath/tools"
+thisinstallerdistro=amazonlinux
+repobased=false
+gitscriptname="installpsh-amazonlinux.psh"
+
+echo
+echo "*** PowerShell Core Development Environment Installer $VERSION for $thisinstallerdistro"
+echo "***    Current PowerShell Core Version: $currentpshversion"
+echo "***    Original script is at: $gitreposcriptroot/$gitscriptname"
+echo
+echo "*** Arguments used: $*"
+echo
+
+# Let's quit on interrupt of subcommands
+trap '
+  trap - INT # restore default INT handler
+  echo "Interrupted"
+  kill -s INT "$$"
+' INT
+
+lowercase(){
+    echo "$1" | sed "y/ABCDEFGHIJKLMNOPQRSTUVWXYZ/abcdefghijklmnopqrstuvwxyz/"
+}
+
+OS=`lowercase \`uname\``
+KERNEL=`uname -r`
+MACH=`uname -m`
+
+if [ "${OS}" == "windowsnt" ]; then
+    OS=windows
+    DistroBasedOn=windows
+    SCRIPTFOLDER=$(dirname $(readlink -f $0))
+elif [ "${OS}" == "darwin" ]; then
+    OS=osx
+    DistroBasedOn=osx
+    # readlink doesn't work the same on macOS
+    SCRIPTFOLDER=$(dirname $0)
+else
+    SCRIPTFOLDER=$(dirname $(readlink -f $0))
+    OS=`uname`
+    if [ "${OS}" == "SunOS" ] ; then
+        OS=solaris
+        ARCH=`uname -p`
+        OSSTR="${OS} ${REV}(${ARCH} `uname -v`)"
+        DistroBasedOn=sunos
+    elif [ "${OS}" == "AIX" ] ; then
+        OSSTR="${OS} `oslevel` (`oslevel -r`)"
+        DistroBasedOn=aix
+    elif [ "${OS}" == "Linux" ] ; then
+        if [ -f /etc/redhat-release ] ; then
+            DistroBasedOn='redhat'
+            DIST=`cat /etc/redhat-release |sed s/\ release.*//`
+            PSUEDONAME=`cat /etc/redhat-release | sed s/.*\(// | sed s/\)//`
+            REV=`cat /etc/redhat-release | sed s/.*release\ // | sed s/\ .*//`
+        elif [ -f /etc/system-release ] ; then
+            DIST=`cat /etc/system-release |sed s/\ release.*//`
+            PSUEDONAME=`cat /etc/system-release | sed s/.*\(// | sed s/\)//`
+            REV=`cat /etc/system-release | sed s/.*release\ // | sed s/\ .*//`
+            if [[ $DIST == *"Amazon Linux"* ]] ; then
```

Comment:
The code to check current OS and `DistroBasedOn` is different from other `installsh-` scripts targeting other Linux distros. Are you going to unify them all?

Reply:
Although it could be done, it's not absolutely necessary so I tacked toward minimal changes to anything I wasn't going to unit test before doing the PR.  install-powershell.sh needs it to correctly call installpsh-amazonlinux.sh and then installpsh-amazonlinux.sh needs it to validate the correct installer.  None of the others need the amazonlinux piece - but I can also understand keeping this bit of code the same.  I don't currently have the bandwidth to test if I change the others.

Let me know what you think we should do with this and then I'll do one update and include the other two items you mentioned in this PR.

Reply:
Thanks for the clarification. I'm fine with the current change.
It would be more consistent if this piece of code can be unified, but that can be done in separate PRs.